closer.go

```diff
@@ -448,16 +473,31 @@ func (c *closer) CloserWaitChan(ctx context.Context) <-chan error {
 	return waitChan
 }
 
+// Implements the Closer interface.
+func (c *closer) RunCloserRoutine(f func() error) {
+	c.CloserAddWait(1)
+	go func() {
+		// CloserAddWait will also add to a closed closer. Ensure we are not in a closing state.
+		if c.IsClosing() {
```

Comment:
This is an "optimization", right? So if the closer has been closed in between the routine being scheduled, we will not run the callback.  
There is still the possibility that after the if check the closer is closed. Maybe the check can be omitted in favor of the code being more simple to read? 

! Subjective Feedback !